Framework/PythonInterface/plugins/algorithms/CropWorkspaceRagged.py

```diff
@@ -0,0 +1,122 @@
+        xmins = self.getProperty('XMin').value
+        xmaxs = self.getProperty('XMax').value
+
+        if len(xmins) == 1 and len(xmaxs) == 1:
+            CropWorkspace(InputWorkspace=inputWS, OutputWorkspace=outputWS,
+                          XMin=xmins[0], XMax=xmaxs[0])
+        else:
+            numSpec = inputWS.getNumberHistograms()
+
+            # fill out the values for min and max as appropriate
+            # numpy 1.7 (on rhel7) doesn't have np.full
+            if len(xmins) == 0:
+                xmins = np.array([Property.EMPTY_DBL]*numSpec)
+            elif len(xmins) == 1:
+                xmins = np.array([xmins[0]]*numSpec)
+            if len(xmaxs) == 0:
+                xmaxs = np.array([Property.EMPTY_DBL]*numSpec)
+            elif len(xmaxs) == 1:
+                xmaxs = np.array([xmaxs[0]]*numSpec)
+
+            # replace nan with EMPTY_DBL in xmin/xmax
+            indices = np.where(np.isnan(xmins))
+            xmins[indices] = Property.EMPTY_DBL
+            indices = np.where(np.isnan(xmaxs))
+            xmaxs[indices] = Property.EMPTY_DBL
+
+            self.log().information('MIN: ' + str(xmins))
+            self.log().information('MAX: ' + str(xmaxs))
+
+            # temporary workspaces should be hidden
+            names = ['__{}_spec_{}'.format(outputWS, i) for i in range(len(xmins))]
+
+            # how much the progress bar moves forward for each spectrum
+            progStep = float(1)/float(2*numSpec)
+
+            # crop out each spectra and conjoin to a temporary workspace
+            accumulationWS = None
+            for i, (name, xmin, xmax) in enumerate(zip(names, xmins, xmaxs)):
+                # don't  go beyond the range of the data
+                x = inputWS.readX(i)
+                if xmin < x[0]:
+                    xmin = Property.EMPTY_DBL
+                if xmax > x[-1]:
+                    xmax = Property.EMPTY_DBL
+
+                progStart = 2 * i * progStep
+
+                # extract the range of the spectrum requested
+                CropWorkspace(InputWorkspace=inputWS, OutputWorkspace=name,
+                              StartWorkspaceIndex=i, EndWorkspaceIndex=i, Xmin=xmin, XMax=xmax,
+                              startProgress=progStart, endProgress=(progStart + progStep),
+                              EnableLogging=False)
+
+                # accumulate
+                if accumulationWS is None:
+                    accumulationWS = name # messes up progress during very first step
+                else:
+                    ConjoinWorkspaces(InputWorkspace1=accumulationWS,
+                                      InputWorkspace2=name,
+                                      startProgress=(progStart + progStep), endProgress=(progStart + 2 * progStep),
+                                      EnableLogging=False)
+
+            RenameWorkspace(InputWorkspace=accumulationWS, OutputWorkspace=outputWS, EnableLogging=False)
```

Comment:
With this line, when running the algorithm with `StoreInADS=False` the output workspace will still show up in the ADS. One solution would be to replace this line by
```python
outputWS = mtd[accumulationWS]
mtd.remove(accumulationWS)
```

and at the end of `PyExec()`:
```python
self.setProperty('OutputWorkspace', outputWS)
```

Reply:
This one created an invalid pointer.

Reply:
Could you check if the attached diff would solve the StoreInADS problem? It also removes a few commented-out lines of leftover code. The file is not compressed, just remove the `.zip` extension.

[CropWorkspaceRagged_fix_StoreInADS.diff.zip](https://github.com/mantidproject/mantid/files/1625489/CropWorkspaceRagged_fix_StoreInADS.diff.zip)